src/packagedcode/swift.py

```diff
@@ -0,0 +1,283 @@
+
+TRACE = SCANCODE_DEBUG_PACKAGE
+
+
+def logger_debug(*args):
+    pass
+
+
+logger = logging.getLogger(__name__)
+
+if TRACE:
+    import sys
+
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+    def logger_debug(*args):
+        return logger.debug(" ".join(isinstance(a, str) and a or repr(a) for a in args))
+
+
+class SwiftManifestJsonHandler(models.DatafileHandler):
+    datasource_id = "swift_package_manifest_json"
+    path_patterns = ("*/Package.swift.json",)
+    default_package_type = "swift"
+    default_primary_language = "swift"
+    description = "json dump of swift manifest"
+    documentation_url = "https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html"
+
+    @classmethod
+    def _parse(cls, swift_manifest, package_only=False):
+
+        if TRACE:
+            logger_debug(
+                f"SwiftManifestJsonHandler: manifest: package: {swift_manifest}"
+            )
+
+        dependencies = get_dependencies(swift_manifest.get("dependencies"))
+
+        package_data = dict(
+            datasource_id=cls.datasource_id,
+            type=cls.default_package_type,
+            primary_language=cls.default_primary_language,
+            namespace=None,
+            name=swift_manifest.get("name"),
+            dependencies=dependencies,
+        )
+
+        return models.PackageData.from_data(package_data, package_only)
+
+    @classmethod
+    def parse(cls, location, package_only=False):
+        with io.open(location, encoding="utf-8") as loc:
+            swift_manifest = json.load(loc)
+
+        yield cls._parse(swift_manifest, package_only)
+
+    @classmethod
+    def assemble(
+        cls,
+        package_data,
+        resource,
+        codebase,
+        package_adder=models.add_to_package,
+    ):
+        """
+        Use the dependencies from `Package.resolved` to create the
+        top-level package with resolved dependencies.
+        """
+        siblings = resource.siblings(codebase)
+        swift_resolved_package_resource = [
+            r for r in siblings if r.name == "Package.resolved"
+        ]
+        dependencies_from_manifest = package_data.dependencies
+
+        processed_dependencies = []
+        if swift_resolved_package_resource:
+            swift_resolved_package_resource = swift_resolved_package_resource[0]
+            swift_resolved_package_data = swift_resolved_package_resource.package_data
+
+            for package in swift_resolved_package_data:
+                version = package.get("version")
+                name = package.get("name")
+
+                purl = PackageURL(
+                    type=cls.default_package_type, name=name, version=version
+                )
+                processed_dependencies.append(
+                    models.DependentPackage(
+                        purl=purl.to_string(),
+                        scope="install",
```

Comment:
Let's use the exact name used upstream for the scope which is "dependencies".
```suggestion
                        scope="dependencies",
```